eth/block_from_raw.go

```diff
@@ -130,48 +130,40 @@ func (b *Block) FromRaw(input string) error {
 	}
 
 	// Difficulty
-	if q, err := NewQuantity(header[7].String); err == nil {
+	if q, err := NewQuantityFromRLP(header[7]); err == nil {
 		b.Difficulty = *q
 	} else {
 		return errors.Wrap(err, "could not convert header field 7 to Difficulty")
 	}
 
-	// Number has a special case where genesis block number is null instead of 0
-	if header[8].String == "0x" {
-		q := QuantityFromUInt64(0)
-		b.Number = &q
+	// Number
+	if q, err := NewQuantityFromRLP(header[8]); err == nil {
```

Comment:
If I understand correctly, the old code doesn't treat `"0x"` as an error, but should, since it's not a valid `QUANTITY`, and `NewQuantityFromRLP()` does treat it as an error.

Reply:
Sort of, hopefully I can explain w/o confusing myself. :)

The issue is that `0x` is valid "RLP" (more specifically, it's valid representation of an empty RLP value when serialized to JSON as part of the eth jsonrpc "schema") but it's not a valid representation of `QUANTITY`, so `NewQuantityFromRLP()` handles that case and converts `rlp.Value{String: "0x"}` to `NewQuantity("0x0")`.

The issue comes from this part of the yellow paper:

https://ethereum.github.io/yellowpaper/paper.pdf

![image](https://user-images.githubusercontent.com/53520/217959202-fc31baa7-f1d1-4f96-9f22-0c82d7e0998a.png)

The "shortest byte array whose big-endian interpretation is the scalar" for ZERO is actually `0x80` in RLP, which is the encoding of an empty byte.  However that at the JSONRPC level `0x` is used to represent an empty byte array (e.g. if you `eth_call` a method that doesn't return the JSONRPC response will be `{ ... "result": "0x"}`.


But this directly conflicts with the JSONRPC `QUANTITY` schema: https://ethereum.org/en/developers/docs/apis/json-rpc/#quantities-encoding

![image](https://user-images.githubusercontent.com/53520/217959869-430d13bb-218b-4e43-9e19-7e57363971c0.png)

So `NewQuantityFromRLP` handles all these corner cases, and should've been used here once it was added.